go/kbnm/hostmanifest/whitelist.go

```diff
@@ -0,0 +1,61 @@
+package hostmanifest
+
+import (
+	"encoding/json"
+	"os"
+	"os/user"
+	"path/filepath"
+)
+
+// whitelistPath is used for installing the whitelist as a JSON into a given
+// path on disk.
+type whitelistPath struct {
+	// Root is the path of the machine-global NativeMessage whitelists.
+	Root string
+	// Home is the path of the NativeMessage whitelists for regular users.
+	Home string
+	// Overlay is an optional path to prefix all paths with. Useful for
+	// generating whitelists for Linux distribution packaging.
+	Overlay string
+}
+
+func (w *whitelistPath) path(u *user.User, appID string) string {
+	if u.Uid == "0" {
+		return filepath.Join(w.Overlay, w.Root, appID+".json")
+	}
+	return filepath.Join(w.Overlay, u.HomeDir, w.Home, appID+".json")
```

Comment:
Is there any scenario where we want home-dir-relative paths and an overlay/prefix?